main-app/src/main/resources/python/submit_job.py

```diff
@@ -125,16 +126,23 @@ def write_driver(driver_spec_loc, driver_spec):
         config_option_overrides = input_params.get("conf")
         if config_option_overrides:
             spark_config.get("pyspark", {}).get("options", {}).get("--conf", {}).update(config_option_overrides)
-        driver_template = get_driver_template("/app/conf/driverTemplate.yaml")
 
-        # variable replace and write new driver podspec
-        # TODO better job of generalizing
-        driver_variables = {
-            'CONTROLLER_NAME': os.environ.get('POD_NAME', "None"),
-            'CONTROLLER_UID': os.environ.get('POD_UID', "None")
-        }
-        driver_spec = replace_template_variables(driver_template, driver_variables)
-        write_driver("driverSpec.yaml", driver_spec)
+        isClusterMode = spark_config.get("pyspark", {}).get("options", {}).get("--conf", {}).get(
+            "spark.kubernetes.driver.master")
+        if isClusterMode:
+            driver_template = get_driver_template("/app/conf/driverTemplate.yaml")
```

Comment:
I sometimes keep this config option set even though I'm only running Spark in local mode, and I think this might bomb if I try running as a local container with the `spark-conf.json` file loaded from a mounted volume. (ex: https://github.com/CognitiveScale/cortex-profiles-sdk-examples/tree/main/bigquery-connection#run-locally-in-a-docker-container-with-spark-submit)

The examples are set up so users shouldn't include this option when running something locally, but I think it would help out to add some error handling if this file doesn't exist so the script just continues. 

Or since the `driverTemplate.yaml` file is kinda small we could it include its current content as a multi-line string and use it as a default. Users can modify the `driverTemplate.yaml` if they want, and there is a still a default in case the file isn't there

Reply:
going off of --deploy-mode now, this will only be set when running in a cluster